packages/core/src/services/documents/modify.test.ts

```diff
@@ -0,0 +1,131 @@
+      commit: commit1,
+      path: 'doc1',
+      content: 'Doc 1 commit 1',
+    })
+    await mergeCommit({ commitId: commit1.id })
+
+    const { commit: commit2 } = await ctx.factories.createDraft({ project })
+
+    await modifyExistingDocument({
+      commitId: commit2.id,
+      documentUuid: doc.documentUuid,
+      content: 'Doc 1 commit 2',
+    }).then((r) => r.unwrap())
+
+    const changedDocuments = await listCommitChanges({
+      commitId: commit2.id,
+    }).then((r) => r.unwrap())
+
+    expect(changedDocuments.length).toBe(1)
+    expect(changedDocuments[0]!.path).toBe('doc1')
+    expect(changedDocuments[0]!.content).toBe('Doc 1 commit 2')
+  })
+
+  it('modifies a document that was created in the same commit', async (ctx) => {
+    const { project } = await ctx.factories.createProject()
+    const { commit } = await ctx.factories.createDraft({ project })
+    const { documentVersion: doc } = await ctx.factories.createDocumentVersion({
+      commit: commit,
+      path: 'doc1',
+      content: 'Doc 1 v1',
+    })
+
+    await modifyExistingDocument({
+      commitId: commit.id,
+      documentUuid: doc.documentUuid,
+      content: 'Doc 1 v2',
+    }).then((r) => r.unwrap())
+
+    const changedDocuments = await listCommitChanges({
+      commitId: commit.id,
+    }).then((r) => r.unwrap())
+
+    expect(changedDocuments.length).toBe(1)
+    expect(changedDocuments[0]!.path).toBe('doc1')
+    expect(changedDocuments[0]!.content).toBe('Doc 1 v2')
+  })
+
+  it('modifying a document creates a change to all other documents that reference it', async (ctx) => {
+    const { project } = await ctx.factories.createProject()
+    const { commit: commit1 } = await ctx.factories.createDraft({ project })
+    const { documentVersion: referencedDoc } =
+      await ctx.factories.createDocumentVersion({
+        commit: commit1,
+        path: 'referenced/doc',
+        content: 'The document that is being referenced',
+      })
+    await ctx.factories.createDocumentVersion({
+      commit: commit1,
+      path: 'unmodified',
+      content: '<ref prompt="referenced/doc" />',
+    })
+    await mergeCommit({ commitId: commit1.id })
+
+    const { commit: commit2 } = await ctx.factories.createDraft({ project })
+
+    await modifyExistingDocument({
+      commitId: commit2.id,
+      documentUuid: referencedDoc.documentUuid,
+      content: 'The document that is being referenced v2',
+    }).then((r) => r.unwrap())
+
+    const changedDocuments = await listCommitChanges({
+      commitId: commit2.id,
+    }).then((r) => r.unwrap())
+
+    expect(changedDocuments.length).toBe(2)
+    expect(
+      changedDocuments.find((d) => d.path === 'referenced/doc'),
+    ).toBeDefined()
+    expect(changedDocuments.find((d) => d.path === 'unmodified')).toBeDefined()
+  })
+
+  it('renaming a document creates a change to all other documents that reference it', async (ctx) => {
```

Comment:
also to all its children given you have to change the paths

Reply:
This is just testing a change of the path of a single document. This does not affect any other document.

The action of renaming a folder will be done as a different service, where it will rename all paths with the same folder.